### PR TITLE
Rails 5.2

### PR DESCRIPTION
**Pull request by thomasdziedzic**
- [ ] someone that knows how production works should review and merge this commit https://github.com/rubygems/rubygems.org/commit/9f740431fd56bedec40543674245c48ae7b28622 or tell me what needs to get merged.

### CODE REVIEW

**Comment by thomasdziedzic** [comment]
I have been able get it down to a single test failure: `bundle exec rails test test/functional/api/v1/rubygems_controller_test.rb -n '/with a confirmed user authenticated On POST to create for new gem/'`

But I don't know why there aren't any full stack traces available for the test.

**Comment by thomasdziedzic** [comment]
I opened an [issue](https://github.com/rubygems/rubygems.org/issues/1773) to track the default configuration migration. I decided not to complete that in this pull request after discovering that the previous rails 5.0 & 5.1 migrations didn't migrate to those default configurations. I think the default configurations should be completed in order, starting with 5.0 and moving forward.

**Comment by greysteil** [comment]
I'm on the Bundler team, not the Rubygems team, but this looks pretty reasonable to me.

One requirement before merging will be to disable encrypted cookies (without which we wouldn't be able to roll back). See https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-5-1-to-rails-5-2 for details.

Because of Rubygems' vendoring setup, we'll also need a maintainer to follow the steps [here](https://github.com/rubygems/rubygems.org/wiki/Updating-Bundled-Gems) when this is ready to merge. @dwradcliffe are you the right person do that?

**Comment by thomasdziedzic** [comment]
Encrypted cookies are disabled by default. This is why the line to add them in is commented out in the new rails configuration file. https://github.com/rubygems/rubygems.org/pull/1771/files#diff-0b94a9ee8eb607505aa56304e7a6f626R20

**Comment by greysteil** [comment]
You're quite right - docs confirm it [here](https://github.com/rails/rails/blob/5-2-1/guides/source/configuring.md#configuring-action-dispatch).

OK, one thing that would be helpful is to get these commits squashed.

File: config/cable.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+development:
+  adapter: async
+
+test:
+  adapter: async
+
+production:
+  adapter: redis
+  url: <%= ENV.fetch("REDIS_URL") { "redis://localhost:6379/1" } %>
+  channel_prefix: gemcutter_production
```
Comment (dwradcliffe) [inline config/cable.yml]: We don't use actioncable, do we need this file? Better to remove and keep things simple or leave it because it's vanilla rails?

**Comment by sonalkr132** [comment]
Merged as 3538e87aa1c6cf58d9326047491e82bfb44ed463. Thanks @thomasdziedzic :heart: :yellow_heart: :green_heart: :blue_heart: 

> Better to remove and keep things simple or leave it because it's vanilla rails?

Removed on master.

**Pull request closed without merging**